File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    let result = 0;
    
    for(let spell of babbling){
        let arr = [0, 0, 0, 0];
        while(spell.length > 0){
            if(spell.slice(0,3) === 'aya'){
                if(arr[0] === 0){
                    spell = spell.slice(3);
                    arr = [1, 0, 0, 0];
                    continue;
                }  
            } else if(spell.slice(0,2) === 'ye'){
                if(arr[1] === 0){
                    spell = spell.slice(2);
                    arr = [0, 1, 0, 0];
                    continue;
                }
            } else if(spell.slice(0,3) === 'woo'){
                if(arr[2] === 0){
                    spell = spell.slice(3);
                    arr = [0, 0, 1, 0];
                    continue;
                }
            } else if(spell.slice(0,2) === 'ma'){
                if(arr[3] === 0){
                    spell = spell.slice(2);
                    arr = [0, 0, 0, 1];
                    continue;
                }
            } 
            break;
        }
        if(spell.length === 0){
            result++;
        }
        
    }
    return result;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    babbling_list = ["aya", "ye", "woo", "ma"]
    
    ans = 0
    
    for i in babbling:
        
        for b in babbling_list:
            i = i.replace(b*2, 'X')
            
        for b in babbling_list:
            i = i.replace(b, ' ')
    
        if i.strip() == '':
            ans += 1
            
    return ans
        